ivy/src/main/scala/sbt/internal/librarymanagement/IvyScalaUtil.scala

```diff
@@ -144,7 +144,13 @@ object IvyScalaUtil {
       val depBinaryVersion = CrossVersion.binaryScalaVersion(id.getRevision)
       def isScalaLangOrg = id.getOrganisation == scalaOrganization
       def isScalaArtifact = scalaArtifacts.contains[String](id.getName)
-      def hasBinVerMismatch = depBinaryVersion != scalaBinaryVersion
+
+      def hasBinVerMismatch =
+        depBinaryVersion != scalaBinaryVersion &&
+          // scala 2.13 is compatible with scala 3.x
+          Seq(depBinaryVersion, scalaBinaryVersion)
+            .forall(bv => bv.startsWith("3") || bv.startsWith("2.13"))
```

Comment:
Looks like dependency-management/scala-organization-version-check is failing. Maybe the logic is flipped here.

Reply:
https://github.com/sbt/librarymanagement/pull/373